src/components/PlayerNotesModal/PlayerNotesModal.tsx

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 2012-2021  Online-Go.com
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import * as React from "react";
+import {_} from 'translate';
+import {Modal, openModal} from "Modal";
+import * as data from "data";
+
+interface Events {
+}
+
+interface PlayerNotesModalProperties {
+    playerId: number;
+}
+
+export class PlayerNotesModal extends Modal<Events, PlayerNotesModalProperties, any> {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            notes: undefined
+        };
+    }
+
+    componentDidMount = () => {
+        super.componentDidMount(); /* this.close() doesn't work if you don't do this */
+        this.setState({ notes: data.get(`player-notes.${this.props.playerId}`) });
+    }
+
+    updateNotes = (ev) => {
+        const new_notes = ev.target.value;
+        if (new_notes.length < 5000) {
+            this.setState({notes: ev.target.value});
+        }
+    }
+
+    saveNotes = () => {
+        data.set(`player-notes.${this.props.playerId}`, this.state.notes, data.Replication.REMOTE_OVERWRITES_LOCAL);
```

Comment:
Here, maybe we should trim `this.state.notes`, and if it's an empty string, do a `data.remove` instead?

Reply:
We could.   Interestingly, it  seems it would make no difference to the functionality at this point -  "" evaluates to false anyhow, so it appears to all intents and purposes like the person "has no notes".   So one could argue it would be "more code to achieve the same thing"...

Reply:
It just makes it so the key will be removed eventually, tidier db / replication, not that big of a deal